tests/integration/mixer/white_black_listing_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"testing"
+	"time"
+
+	"fmt"
+	"net/http"
+	"os"
+
+	"fortio.org/fortio/fhttp"
+	"fortio.org/fortio/periodic"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+func TestWhiteListing(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		bookinfoNs, err := namespace.New(ctx, "istio-bookinfo", true)
+		if err != nil {
+			t.Fatalf("Could not create istio-bookinfo Namespace; err:%v", err)
+		}
+		d := bookinfo.DeployOrFail(t, ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookInfo})
+		g := galley.NewOrFail(t, ctx, galley.Config{})
+		_ = mixer.NewOrFail(t, ctx, mixer.Config{Galley: g})
+
+		g.ApplyConfigOrFail(
+			t,
+			d.Namespace(),
+			bookinfo.NetworkingBookinfoGateway.LoadOrFailGatewayFileWithNamespace(t, bookinfoNs.Name()))
+		g.ApplyConfigOrFail(
+			t,
+			d.Namespace(),
+			bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadOrFailWithNamespace(t, bookinfoNs.Name()),
+			bookinfo.NetworkingVirtualServiceAllV1.LoadOrFailWithNamespace(t, bookinfoNs.Name()),
+		)
+
+		// Wait for rules to sync.
+		time.Sleep(time.Second * 30)
```

Comment:
We need a better solution for this in general, but 30s seems like a long time?

Reply:
I think that's within historical data. Lack of signals on change propagation is a systemic issue in istio, so this PR is not doing anything special.

Reply:
From other tests I see 3, 10, 10, 1, 5, 1, 3, 1, 5, 30 seconds.

I am just worried our test run time gets crazy long as we add more tests

Reply:
At times this 30s is also not enough.. I am sending 30 requests below, just to make sure rules get synced and I see a 403. 

Yet to test but I was thinking what if we query for things like "kubectl get instance -n istio-bookinfo-1-63730" and wait for instance to show up?

Reply:
@howardjohn for this rule to be effective we need Pilot, Mixer, and all the proxies by synced up. Other tests might just need one proxy to be up to date. So it's fairly reasonable for any Mixer test to require longer config propagation.

Reply:
The CRD is probably applied to kube very quickly (it maybe even blocks until it is created, not sure). I think the problem is the propagation to Mixer - not sure how mixer works at all though so maybe not

Reply:
@gargnupur The real signal is waiting for Mixer to apply the config. That means Galley picks it up, and Mixer sees it. I think Mixer has trouble starting due to failing to reach Galley at the beginning so there's some stabilization delay. Checking that k8s is does not say much about whether Galley pushed it to all Mixers.

Reply:
:( not for this PR but later we can use mixer logs to see CRD's are synced?

Reply:
Why can't we use an exponential back-off for trying to get a 200 in place of the `30s` upfront?  I'm not familiar with the new framework, but surely we have a `Retry` or equivalent exposed somewhere, right?

Then, for the Mixer one, we can use Mixer's self-monitoring metrics in a similar fashion. Poll, and watch for the change in number of instances (or whatever signal we want).

Reply:
Thanks for the suggestion.. added a retry to wait till specified status code..

Reply:
Self-monitoring metric seems like a great idea. We should pursue that idea in a follow-up.

Reply:
Agreed.. will followup in a different PR